wiki-morty/src/components/Detail/Detail.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
+
+class Detail extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {  
+      isLoaded: false
+    }
+  }
+  
+  componentDidMount() {
+    const path = this.props.match.url.split('/');
+    let type = path[1].slice(0,-1)
+    fetch(`https://rickandmortyapi.com/api/${type}/${path[2]}`)
+    .then(response => response.json())
+    .then(json => this.setState({
+      result: json,
+      type: type,
+      isLoaded: true
+    }))
+    .catch()
+  }
+
+
+
+  render() { 
+    const isLoaded = this.state.isLoaded;
+    let content;
+    if (!isLoaded) {
+      content = <div>Loading....</div>
+    } else {
+      if (this.state.type == 'character') {
+        content = 
```

Comment:
This would also be a good case to use a switch statement + separate component for each type.